### PR TITLE
Add GPv2Settlement contract hook

### PR DESCRIPTION
**Pull request by Velenir**
- Adds abi (tentative), to be updated to keep in sync with contracts
- Interface
- Contract

Also good idea may be to import ABI from npm, but it will likely pull in more thanjust abi (not sure how tree-shaking will handle it)
But for now this mimicks how uniswap adds abis

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
   * **🎩 [Dapp](https://pr51--gpv2swap.review.gnosisdev.com)**: Testing web app

**Review by W3stside — APPROVED** [review]

File: src/custom/constants/GPv2Settlement.json [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        "type": "bytes"
+      }
+    ],
+    "name": "invalidateOrder",
+    "outputs": [],
+    "stateMutability": "nonpayable",
+    "type": "function"
+  },
+  {
+    "inputs": [
```
Comment (anxolin) [inline src/custom/constants/GPv2Settlement.json:108]: Do we want all the ABI?

File: src/custom/constants/GPv2Settlement.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { ChainId } from '@uniswap/sdk'
+import { Interface } from '@ethersproject/abi'
+// Can even prune ABI down to onlywhat we use
+// like events only
+import GP_V2_SETTLEMENT from './GPv2Settlement.json'
```
Comment (anxolin) [inline src/custom/constants/GPv2Settlement.ts:5]: Yes, I would do that. Our use of it is minimal, I wouldn't add extraweight to the web if we can avoid it

File: src/custom/constants/GPv2Settlement.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+export const GP_V2_SETTLEMENT_INTERFACE = new Interface(GP_V2_SETTLEMENT)
+
+export const GP_SETTLEMENT_CONTRACT_ADDRESS: Partial<Record<ChainId, string>> = {
+  [ChainId.MAINNET]: '0x6F400810b62df8E13fded51bE75fF5393eaa841F',
+  [ChainId.RINKEBY]: '0xC576eA7bd102F7E476368a5E98FA455d1Ea34dE2'
```
Comment (anxolin) [inline src/custom/constants/GPv2Settlement.ts:11]: you can update to develop now and it should be there this constants

File: src/hooks/useContract.ts [hunk 4]
```diff
@@ -24,7 +24,7 @@ import { getContract } from '../utils'
 import { useActiveWeb3React } from './index'
 
 // returns null on errors
-function useContract(address: string | undefined, ABI: any, withSignerIfPossible = true): Contract | null {
+export function useContract(address: string | undefined, ABI: any, withSignerIfPossible = true): Contract | null {
```
Comment (anxolin) [inline src/hooks/useContract.ts:27]: src modification, but a `fair` one to do :)

**Review by anxolin — COMMENTED** [review]
> Also good idea may be to import ABI from npm, but it will likely pull in more thanjust abi (not sure how tree-shaking will handle it)

I agree, I wouldn't use the full ABI. let's reduced it to just the event we want if possible
Maybe, also the one to get the allowance manager if we plan to use that. Although I think using the library should be also good...

**Comment by Velenir** [comment]
> I agree, I wouldn't use the full ABI. let's reduced it to just the event we want if possible
> Maybe, also the one to get the allowance manager if we plan to use that. Although I think using the library should be also good...

Done, 0428b7d Only left `event Trade and allowanceManager`

**Review by anxolin — APPROVED** [review]

**Pull request merged**